hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnection.java

```diff
@@ -240,24 +240,38 @@ public void close(String reason, Throwable cause) {
             logger.warning(e);
         }
 
-        Object connAddress = getConnectionAddress();
-        String message = "Connection [" + connAddress + "] lost. Reason: ";
-        if (closeReason != null) {
-            message = closeReason;
-        } else if (cause == null) {
-            message += "Socket explicitly closed";
-        } else {
-            message += cause.getClass().getName() + "[" + cause.getMessage() + "]";
-        }
-
-        logger.info(message);
         connectionManager.onClose(this);
         connectionManager.getIoService().onDisconnect(endPoint, cause);
         if (cause != null && monitor != null) {
             monitor.onError(cause);
         }
     }
 
+    private void logClose() {
+        String message = toString() + " closed. Reason: ";
+        if (closeReason != null) {
+            message += closeReason;
+        } else if (closeCause != null) {
```

Comment:
Hmm. I missed `warning` log part... 
Still, having exception in 1st log line may be useful, for instance while doing a simple `grep`.. but not essential...